File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const arr = []
    
    let day = 1;
    let count = 0;
    while (progresses.length > 0) {
        const progress = progresses[0] + (speeds[0] * day)
        
        if (progress >= 100) {
            progresses.shift();
            speeds.shift();
            count++;
        } else {
            day++;
            if (count > 0) {
                arr.push(count);
                count = 0;
            }
        }
    }
    
    arr.push(count);
    
    return arr;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> days = new LinkedList<>();
        
        for (int i = 0; i < progresses.length; i++) {
            int day = (int) Math.ceil(
                (double) (100 - progresses[i]) / speeds[i]
            );
            days.add(day);
        }
        
        int count = 1;
        int day = days.remove();
        
        List<Integer> answer = new ArrayList<>();
        while(days.size() > 0) {
            int current = days.remove();
            if (current <= day) {
                count++;
            } else {
                day = current;
                answer.add(count);
                count = 1;
            }
        }
        answer.add(count);
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}